### PR TITLE
Immediately acknowledge GRPC subscriptions

### PR DESCRIPTION
**Pull request by ma2bd**
## Motivation

We observe timeouts in GRPC clients when subscribing to a Linera proxy behind Nginx instances.

## Proposal

Make sure the GRPC `subscribe` immediately returns an empty value. This should force completion of the HTTP2 header and prevent the bug

## Test Plan

* CI should catch any regression
* will need to be manually tested

## Release Plan

- These changes should be backported to the latest `devnet` branch, then
    - be released in a new SDK,
    - be released in a validator hotfix.
- These changes should be backported to the latest `testnet` branch, then
    - be released in a new SDK,
    - be released in a validator hotfix.

### CODE REVIEW

File: linera-rpc/src/grpc/client.rs [hunk 1]
```diff
@@ -279,7 +279,13 @@ impl ValidatorNode for GrpcClient {
                     true
                 })
             })
-            .filter_map(|result| future::ready(result.ok()));
+            .filter_map(|result| {
```
Comment (jvff) [inline linera-rpc/src/grpc/client.rs:282]: Maybe not in this PR, but it feels like we should at least `debug` log any errors :thinking:

File: linera-rpc/src/grpc/conversions.rs [hunk 2]
```diff
@@ -903,6 +907,13 @@ pub mod tests {
                 hash: CryptoHash::new(&Foo("".into())),
             },
         };
-        round_trip_check::<_, api::Notification>(notification);
+        let message = api::Notification::try_from(notification.clone()).unwrap();
```
Comment (jvff) [inline linera-rpc/src/grpc/conversions.rs:910]: Also not a blocker, but this feels suspicious. It feels like we'd like to keep the `round_trip_check` for both cases, but split the new `ack` test into its own `test_empty_notification` test.

Seems okay for now, but we should revisit this test later.

**Review by jvff — APPROVED** [review]

File: linera-rpc/src/grpc/conversions.rs (hunk 2 above)
Comment (ma2bd) [reply linera-rpc/src/grpc/conversions.rs:910]: `round_trip_check` doesn't apply any more.

File: linera-rpc/src/grpc/client.rs (hunk 1 above)
Comment (ma2bd) [reply linera-rpc/src/grpc/client.rs:282]: done

File: linera-rpc/src/grpc/conversions.rs (hunk 2 above)
Comment (deuszx) [inline linera-rpc/src/grpc/conversions.rs:910]: I was told we should be using `expect(<reason>)` rather than unwraps :)

Comment (ma2bd) [reply linera-rpc/src/grpc/conversions.rs:910]: That one is in a test though :)

**Pull request merged**